test/unit-test/core_json_utest.c

```diff
@@ -935,6 +940,100 @@ void test_JSON_Search_Legal_Array_Documents( void )
     doSearch( "[5]", ARRAY_ELEMENT_5_TYPE, ARRAY_ELEMENT_5 );
 }
 
+/**
+ * @brief Test that JSON_Iterate returns the given values from a JSON array.
+ */
+void test_JSON_Iterate_Legal_Array_Documents( void )
+{
+    JSONStatus_t jsonStatus;
+    size_t start = 0, next = 0;
+    JSONPair_t pair = { 0 };
+
+#define doIterate( type, answer )                                    \
```

Comment:
Just a nit about the name
```suggestion
#define iterateObject( type, answer )                                    \
```